chapter/simpleTypes.tex

```diff
@@ -46,100 +46,108 @@ \section{Simply Typed Lambda Calculus}
 \end{mathpar}
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\section{Curry-Howard Isomorphism}
+\section{Curry--Howard Isomorphism and Some New Constructs}
 
-Now, let's remove all terms from typing rules above, replace $\times$ with $\land$,
-$+$ with $\lor$ and squint our eyes a little. What do we see? A wild natural deduction appears!
+Now, let's remove all terms from typing rules above and squint our eyes a little. What do we see? A wild natural deduction appears!
 
 \begin{mathpar}
   \inferrule{\tau \in \Gamma}
             {\Gamma\vdash \tau}
-  
-  \\\\
 
   \inferrule{\Gamma, \tau_1 \vdash \tau_2}
             {\Gamma\vdash \tau_1 \to \tau_2}
   
   \inferrule{\Gamma\vdash \tau_2 \to \tau_1 \\ \Gamma\vdash \tau_2}
             {\Gamma\vdash \tau_1}
 
-  \\\\
-          
-  \inferrule{\Gamma\vdash \tau_1 \\ \Gamma\vdash \tau_2}
-            {\Gamma\vdash \tau_1 \land \tau_2}
-
-  \inferrule{\Gamma\vdash \tau_1 \land \tau_2}
-            {\Gamma\vdash \tau_i}
+\end{mathpar}
+Notice that the structure of a term mirrors its type derivation -- whenever we see a lambda-abstraction, we know that an implication was introduced; whenever we see an application, we know that an implication was eliminated.
```

Comment:
For dash used as colon or parentheses we use em dash `---` instead of en dash `--` (en dash is used to denotes ranges, relationships, etc., for instance `pp. 13--29`). It is also worth noting, that when typesetting in plaintext--e.g. in emails--we write double dash for both en dash and em dash.
Moreover in English typographic rules, em dash is always closed (not surrounded by spaces). This is opposite to Polish typography, where em dash (pol. *myślnik* or *pauza*) is always open.